### PR TITLE
[FLINK-26688] Completely remove unused options from TableConfig

### PR DESCRIPTION
**Pull request by matriv**
Follows: 1e3344854b3bfd7b6f5498a0297ba49d55127003
Follows: 2eacc7373aa0fca2f2c59a72b0affb0ebda49ddf

## What is the purpose of the change

Further updates to clean up TableConfig and its usages

## Brief changelog:

 - Completely remove unused options from `TableConfig`
 - Deprecate `new TableConfig()` in favour of `TableConfig.getDefault()`
 - Remove the `TableConfig` from python's `StreamTableEnviroment#create()` and allow configuration only via `EnviromentSettings.with_configuration()`.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): **no**
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: **no**
  - The serializers: **no**
  - The runtime per-record code paths (performance sensitive): **no**
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: **no**
  - The S3 file system connector: **no**

## Documentation

  - Does this pull request introduce a new feature? **no**
  - If yes, how is the feature documented? **not applicable**

### CODE REVIEW

**Review by twalthr — APPROVED** [review]
LGTM

File: flink-python/pyflink/table/table_environment.py [hunk 1]
```diff
@@ -1668,7 +1668,10 @@ def create(stream_execution_environment: StreamExecutionEnvironment = None,  # t
             >>> table_config = TableConfig()
             >>> table_env = StreamTableEnvironment.create(env, table_config)
             # create with StreamExecutionEnvironment and EnvironmentSettings.
-            >>> environment_settings = EnvironmentSettings.in_streaming_mode()
+            >>> configuration = Configuration()
```
Comment (twalthr) [inline flink-python/pyflink/table/table_environment.py:1671]: `create with StreamExecutionEnvironment and TableConfig.` is outdated

File: flink-python/pyflink/table/table_environment.py [hunk 2]
```diff
@@ -1668,7 +1668,10 @@ def create(stream_execution_environment: StreamExecutionEnvironment = None,  # t
...
+            >>> configuration.set_string('execution.buffer-timeout', '1 min')
+            >>> environment_settings = EnvironmentSettings.new_instance()
```
Comment (twalthr) [inline flink-python/pyflink/table/table_environment.py:1673]: can we improve the indention here e.g. new line before `.with_configuration`

**Review by twalthr — CHANGES_REQUESTED** [review]
Thanks @matriv. I had some comments for the last commits.

File: flink-python/pyflink/table/table_environment.py [hunk 3]
```diff
@@ -1664,11 +1663,13 @@ def create(stream_execution_environment: StreamExecutionEnvironment = None,  # t
...
-            >>> table_config = TableConfig()
-            >>> table_env = StreamTableEnvironment.create(env, table_config)
             # create with StreamExecutionEnvironment and EnvironmentSettings.
-            >>> environment_settings = EnvironmentSettings.in_streaming_mode()
+            >>> configuration = Configuration()
+            >>> configuration.set_string('execution.buffer-timeout', '1 min')
+            >>> environment_settings = EnvironmentSettings
+            >>>     .new_instance()
+            >>>     .in_streaming_mode()
+            >>>     .with_configuration(configuration)
```
Comment (twalthr) [inline flink-python/pyflink/table/table_environment.py:1672]: is there no `build()` method at the end?

**Review by twalthr — CHANGES_REQUESTED** [review]
One minor comment. Should be good in the next iteration.

File: flink-python/pyflink/table/table_environment.py (hunk 3 above)
Comment (matriv) [reply flink-python/pyflink/table/table_environment.py:1672]: Apologies @twalthr, I've rushed. fixed it.

**Pull request closed without merging**